### PR TITLE
Initial pdns policy module

### PR DESCRIPTION
**Pull request by rubenk**
Signed-off-by: Ruben Kerkhof <ruben@rubenkerkhof.com>

### CODE REVIEW

**Comment by doverride** [comment]
Looks like it is actually lacking the "pdns_can_network_connect_db" boolean
declaration?

You might not notice it since it is currently optional

**Comment by rubenk** [comment]
Thanks, I've added it.

File: pdns.te [hunk 1]
```diff
@@ -0,0 +1,84 @@
+policy_module(pdns, 1.0.1)
+
+########################################
+#
+# Declarations
+#
+
+## <desc>
+### <p>
```
Comment (wrabcak) [inline pdns.te]: ## is enough

Comment (rubenk) [inline pdns.te]: What do you mean?

File: pdns.te [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+corenet_tcp_bind_dns_port(pdns_t)
+corenet_udp_bind_dns_port(pdns_t)
+
+files_pid_filetrans(pdns_t, pdns_var_run_t, { file sock_file })
+manage_files_pattern(pdns_t, pdns_var_run_t, pdns_var_run_t)
+manage_sock_files_pattern(pdns_t, pdns_var_run_t, pdns_var_run_t)
+
+auth_use_nsswitch(pdns_t)
+
+dev_read_urand(pdns_t)
```
Comment (wrabcak) [inline pdns.te]: This can be removed, this rule is included in auth_use_nsswitch.

**Comment by wrabcak** [comment]
Hi, sorry for my delay. 
This looks really good! I have just 2 things.
First, If this policy just read from /etc/pdns, we don't need this label, in fedore we allow read etc_t files in default. 
Second thing, If you look on optional block, rules mysql_read_conf and mysql_stream_connect could be in tunable_policy block, due to this is needed when pdns is connecteing to db port? 

Thank you for your policy!

**Comment by rubenk** [comment]
Hi Lukas,

I labelled the config file since it can contain passwords, for example for the mysql backend.

As for your second comment, if I moved the rules to inside the tunable block and the tunable is off, I think that would break if pdns connected locally to mysql via a unix socket.

**Comment by wrabcak** [comment]
Understand. Merging. 
Thank you!

**Pull request merged**